### PR TITLE
[libavif] Fix usage requirements export

### LINKED ISSUE #42112: [libavif] Can't use static build after 1.1.1 update
`CMakeLists.txt`:
```cmake
cmake_minimum_required(VERSION 3.10)
project(HelloWorld)
find_package(libavif CONFIG REQUIRED)
add_executable(HelloWorld helloworld.cpp)
target_link_libraries(HelloWorld PRIVATE avif)
```

`helloworld.cpp`:
```cpp
#include <avif/avif.h>
#include <iostream>

int main() {
    const char* version = avifVersion();
    std::cout << "libavif version: " << version << std::endl;
    return 0;
}
```

`vcpkg.json`:
```json
{
  "builtin-baseline": "e60236ee051183f1122066bee8c54a0b47c43a60",
  "overrides": [
    {
      "name": "libavif",
      "version": "1.1.1"
    }
  ],
  "dependencies": [
    {
      "name": "libavif",
      "features": [
        "dav1d"
      ]
    }
  ]
}
```

When trying to build in macOS results in the following error: `helloworld.cpp:1:10: fatal error: 'avif/avif.h' file not found`. I believe the same would happen in Linux, but not Windows (didn't try w/ this minimal project but that's what I was seeing with a much larger project that uses libavif), with the difference presumably being that the library is static in macOS/Linux. If the override in `vcpkg.json` is set to version `1.0.4#2`, it works fine.

Comparing the CMake files in `vcpkg_installed/arm64-osx/share/libavif` from version `1.0.4#2` to `1.1.1` reveals that the following section disappeared:

```cmake
set_target_properties(avif PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "\$<\$<NOT:\$<CONFIG:DEBUG>>:${VCPKG_IMPORT_PREFIX}/lib/libdav1d.a>;\$<\$<CONFIG:DEBUG>:${VCPKG_IMPORT_PREFIX}/debug/lib/libdav1d.a>;\$<LINK_ONLY:yuv>;\$<LINK_ONLY:m>;\$<LINK_ONLY:Threads::Threads>"
)
```

### PR DESCRIPTION
**Pull request by dg0yt**
Amends #41569.
Fixes https://github.com/microsoft/vcpkg/issues/42112.
Alternative to #42151.

### CODE REVIEW

**Comment by dg0yt** [comment]
CI still doesn't start @BillyONeal.

**Comment by BillyONeal** [comment]
Yeah, something is broken. Having difficulty trying to restore it due to the storm damage last night https://www.seattletimes.com/seattle-news/weather/bomb-cyclone-over-450000-people-without-power-in-western-washington/

**Comment by BillyONeal** [comment]
I destroyed and recreated all the pools and that seems to have fixed it for now, macs will be slow until I wake them all back up since that is a manual process.

**Comment by BillyONeal** [comment]
:sigh: One more time

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/libavif/dependencies.diff [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ 
+@@ -849,13 +850,38 @@ if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
+ 
+     # Enable CMake configs in VCPKG mode
+     if(BUILD_SHARED_LIBS OR VCPKG_TARGET_TRIPLET)
+-        install(EXPORT ${PROJECT_NAME}-config DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
++        install(EXPORT ${PROJECT_NAME}-config DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME} FILE ${PROJECT_NAME}-targets.cmake)
++        if(NOT BUILD_SHARED_LIBS)
++            target_include_directories(avif_static PUBLIC $<INSTALL_INTERFACE:include>)
++            set(avif_static_link_libs yuv)
```
Comment (BillyONeal) [inline ports/libavif/dependencies.diff:21]: Stupid question: Why is this `yuv` rather than `yuv::yuv` like above?

Comment (dg0yt) [reply ports/libavif/dependencies.diff:21]: This port is taking separate paths for static vs. dynamic linkage.

`yuv::yuv` is from libavif's find module, used to link the shared libavif lib. No transitive usage requirements.

This PR just adds dependencies to take effect *downstream* when linking the *static* lib. `yuv` is vcpkg's unofficial target provided by `find_dependency(libyuv CONFIG)`.

Comment (dg0yt) [reply ports/libavif/dependencies.diff:21]: (It is very hard to merge the separate paths. The static libs is merged with custom commands instead using plain cmake.)

**Pull request merged**